src/Apps/Collect/Components/ArtworkGrid/CollectArtworkFilterRefetch.tsx

```diff
@@ -72,57 +74,62 @@ export class CollectArtworkFilterRefetch extends Component<
   }
 }
 
-export const CollectArtworkGridRefreshContainer = createRefetchContainer(
+export const CollectArtworkGridRefetchContainer = createRefetchContainer(
   CollectArtworkFilterRefetch,
-  graphql`
-    fragment CollectArtworkFilterRefetch_query on Query
-      @argumentDefinitions(
-        medium: { type: "String", defaultValue: "*" }
-        major_periods: { type: "[String]" }
-        partner_id: { type: "ID" }
-        for_sale: { type: "Boolean" }
-        at_auction: { type: "Boolean" }
-        aggregations: {
-          type: "[ArtworkAggregation]"
-          defaultValue: [MEDIUM, TOTAL]
-        }
-        sort: { type: "String", defaultValue: "-partner_updated_at" }
-      ) {
-      filter_artworks(aggregations: $aggregations, size: 0) {
-        aggregations {
-          slice
-          counts {
-            name
-            id
-          }
+  {
+    viewer: graphql`
+      fragment CollectArtworkFilterRefetch_viewer on Viewer
+        @argumentDefinitions(
```

Comment:
I'm still a bit confused as to where in the relay container tree we need to include argument definitions.
cc @mzikherman @alloy

Reply:
Yea, the GraphQL spec says all of these declared variables are global, and can be used by all fragments. So you would only need to list it at the top of your query renderer: https://github.com/artsy/reaction/pull/1212/files#diff-36e1f38238a06214205cd08ccc4a670eR12 .

I believe `@arguments` and `@argumentDefinitions` are Relay-added directives (the generated queries do _not_ contain these), and without them I've gotten Relay compiler issues.

So I _think_ (~90% sure) that it's basically a Relay thing/quirk that you need to keep defining/passing them down like this. But I haven't played around with it too much recently.

Reply:
Correct, these are Relay specific directives that allow fragments to define variables, even though normally variables only get defined at the root of your operation. It’s akin to components having a props interface and passing those props along vs using global variables, both work but you lose visibility into them when going down that route. Visibility in this case means e.g. that Relay knows if variables are unused and probably more things.

I’d invite you to read the relevant parts of the Relay source to gain a full understanding, possibly extend the documentation on the topic, and thus sharing this knowledge with us all.